File: 프로그래머스/lv1/12930. 이상한 문자 만들기/이상한 문자 만들기.js

function solution(s) {
    let Box = s.split(" ")  // ["try", "hello", "world"]
    let resultBox = []
    // let resultBox = ``
    for(let i=0; i < Box.length; i++){
        let innerBox =""
        for(let j=0; j < Box[i].length; j++){
            if(j%2 === 0){
                console.log(Box[i][j].toUpperCase())
                innerBox += Box[i][j].toUpperCase() // => T Y
                // innerBox += Box[i][j].toUpperCase() // => T Y
            } else {
                innerBox += Box[i][j].toLowerCase()  // r (만약 대문자면 소문자로)
            }
        } // "TrY"
        resultBox.push(innerBox)  // => ["TrY", "HeLlO", "WoRlD"]
    }
    return resultBox.join(" ")
    // return resultBox.reduce((prev, curv)=> {
    //     `${prev}, ${curv}`
    // })  // 정규 표현식 
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        String[] str = s.split("");
        int index = 0;
        for(int i = 0; i < str.length; i++){
            if(str[i].equals(" ")){
                index = 0; 
                continue;
            }
            if(index % 2 == 0){
                str[i] = str[i].toUpperCase();
            } else {
                str[i] = str[i].toLowerCase();
            }
            index++;
        }
        return String.join("", str);
    }
}
